cpp/oneapi/dal/algo/connected_components/backend/cpu/vertex_partitioning_default_kernel.hpp

```diff
@@ -20,16 +20,159 @@
 #include "oneapi/dal/backend/common.hpp"
 #include "oneapi/dal/backend/memory.hpp"
 #include "oneapi/dal/backend/interop/common.hpp"
+#include "oneapi/dal/table/homogen.hpp"
 #include "oneapi/dal/detail/error_messages.hpp"
+#include "oneapi/dal/backend/primitives/rng/rng_engine.hpp"
 
 namespace oneapi::dal::preview::connected_components::backend {
+using namespace oneapi::dal::preview::detail;
+using namespace oneapi::dal::preview::backend;
+
+inline void link(std::int64_t u, std::int64_t v, std::int32_t *components) {
+    std::int32_t p1;
+    std::int32_t p2;
+    std::int32_t h;
+    std::int32_t l;
+    p1 = components[u];
+    p2 = components[v];
+    while (p1 != p2) {
+        if (p1 > p2) {
+            h = p1;
+            l = p2;
+        }
+        else {
+            h = p2;
+            l = p1;
+        }
+        if (components[h] == h) {
+            components[h] = l;
+            break;
+        }
+        p1 = components[components[h]];
+        p2 = components[l];
+    }
+}
+
+inline void compress(std::int64_t u, std::int32_t *components) {
+    while (components[components[u]] != components[u]) {
```

Comment:
Why while here?? if not equal -> make them equal ->  next while  useless check.

Reply:
seems that it was something like atomic operation - compare and swap, but:
1. it is not atomic in the code.
2. This logic is useless right now since no threading is added.

Reply:
You are right. This will be changed to "if"